### PR TITLE
Test Mermaid fix for corner cases with Exits

### PR DESCRIPTION
**Pull request by mhdirkse**
parent d81780464b0d8cd13336a32724653197af529892
author Martijn Dirkse <martijn@integrationpartners.nl> 1712669736 +0200 committer Martijn Dirkse <martijn@integrationpartners.nl> 1717163720 +0200

Test Mermaid fix for corner cases with Exits

Correct previous commit

Show Mermaid XSLT parameter in log

Properly log parameter XML elements

make success exit even if error exit already exists

Unit tests for Mermaid generation edge cases

Correct test case

Fix previous commit

Use syntax 1 for test adapter

Fix the Mermaid ids

Test implicit success forward if only explicit error forward

Fix ids

Test no implicit error forward created

Test default, no implicits

Remove OpenWeather token

Demote log statements to Debug level

Do not make Mermaid the default yet

### CODE REVIEW

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java [hunk 1]
```diff
@@ -206,10 +208,11 @@ private boolean matchesTestClassPath(MetadataReader reader, MetadataReaderFactor
 	}
 
 	@Override
-	public void generateFlow(String xml, OutputStream outputStream) throws FlowGenerationException {
+	public void generateFlow(String xml, OutputStream outputStream, String name) throws FlowGenerationException {
 		try {
+			log.debug("Generating flow for the following XML: [{}]", xml);
 			String flow = generateMermaid(xml);
-
+			log.debug("Have the following Mermaid code to draw Configuration or Adapter [{}]: {}", name, flow);
```
Comment (nielsm5) [inline core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:215]: Dit is denk ik niet zo handig. Zowel de gehele adapter/configuratie xml loggen, en het meegeven van de naam zou niet moeten hoeven.

Comment (mhdirkse) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:215]: Deze code had ik nodig om de unit tests te kunnen maken. Ik paste telkens mijn voorbeeld config aan. Daarna keek ik of het plaatje lukte. Vervolgens las ik de syntax 1 (oude syntax, input van de test) en de gegenereerder Mermaid (output) uit de log. Ik had toen ik aan het ontwikkelen was log level info omdat ik niet wou uitzoeken hoe ik debug logs kon zien. Toen de tests klaar waren heb ik de logs teruggezet naar level "debug". Normaal gesproken zou je ze dus niet moeten zien zodat de log compact blijft.

Comment (mhdirkse) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:215]: Het was ook erg handig om de naam te hebben. Zo kan je makkelijk zoeken van welke config of adapter je de Mermaid ziet.

Comment (nielsm5) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:215]: Ik snap dat het voor het testen handig is, dan kan het bij de test er bijgezet worden. Dit moet niet in de productie code komen.

Comment (mhdirkse) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java]: Ik heb bijna alle log statements waar we het over hadden weggehaald, maar ik heb er één laten staan. Dat is bij het initialiseren van de MermaidFlowGenerator.java. Kan deze wel?

**Comment by mhdirkse** [comment]
Ik heb alle FF! testen op mijn laptop gedraaid, inclusief de testen die ik in deze PR toevoeg. Ze slagen.

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java (hunk 1 above)
Comment (nielsm5) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java]: Ik heb nog even gekeken naar de configuratie xml, naast dat deze in de console beschikbaar is, wordt deze ook gelogd in de configuration-log.

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java [hunk 2]
```diff
@@ -84,6 +84,7 @@ public MermaidFlowGenerator() {
 	@Override
 	public void afterPropertiesSet() throws Exception {
 		String frankElementsList = compileFrankElementList();
+		log.debug("Initialized frankElementList to be [{}]", frankElementsList);
```
Comment (nielsm5) [inline core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:87]: Misschien is generated beter? Ik zie dat het nuttig is om dit te loggen maar gezien de grote lap tekst die in de log verschijnt is trace net iets handiger.
```suggestion
		log.trace("generated frankElementList [{}]", frankElementsList);
```

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java (hunk 1 above)
Comment (mhdirkse) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java]: Bedankt.

**Review by nielsm5 — APPROVED** [review]
